Mathlib/Data/Nat/PartENat.lean

```diff
@@ -149,18 +149,10 @@ protected theorem casesOn {P : PartENat → Prop} : ∀ a : PartENat, P ⊤ →
   exact PartENat.casesOn'
 #align part_enat.cases_on PartENat.casesOn
 
--- Porting note : The last instance in this file (`LinearOrderedAddCommMonoidWithTop`)
--- causes the linter to complain here because with that instance `simp` could
--- proof `top_add`. Therefore the linter has been silenced here.
-@[nolint simpNF, simp]
 theorem top_add (x : PartENat) : ⊤ + x = ⊤ :=
```

Comment:
```suggestion
-- not a simp lemma since the `LinearOrderedAddCommMonoidWithTop` instance makes it redundant
theorem top_add (x : PartENat) : ⊤ + x = ⊤ :=
```
or similar.

Another option would be to remove the `simp` attribute just before the instance is created.